code/game/gamemodes/revolution/revolution.dm

```diff
@@ -281,73 +152,35 @@
 		return TRUE
 
-/////////////////////////////////////
-//Adds the rev hud to a new convert//
-/////////////////////////////////////
-/datum/game_mode/proc/update_rev_icons_added(datum/mind/rev_mind)
-	var/datum/atom_hud/antag/revhud = GLOB.huds[ANTAG_HUD_REV]
-	revhud.join_hud(rev_mind.current)
-	set_antag_hud(rev_mind.current, ((rev_mind in head_revolutionaries) ? "hudheadrevolutionary" : "hudrevolutionary"))
-
-/////////////////////////////////////////
-//Removes the hud from deconverted revs//
-/////////////////////////////////////////
-/datum/game_mode/proc/update_rev_icons_removed(datum/mind/rev_mind)
-	var/datum/atom_hud/antag/revhud = GLOB.huds[ANTAG_HUD_REV]
-	revhud.leave_hud(rev_mind.current)
-	set_antag_hud(rev_mind.current, null)
-
-//////////////////////////
-//Checks for rev victory//
-//////////////////////////
-/datum/game_mode/revolution/proc/check_rev_victory()
-	for(var/datum/mind/rev_mind in head_revolutionaries)
-		for(var/datum/objective/mutiny/objective in rev_mind.objectives)
-			if(!(objective.check_completion()))
-				return FALSE
-	return TRUE
-
-/////////////////////////////
-//Checks for a head victory//
-/////////////////////////////
-/datum/game_mode/revolution/proc/check_heads_victory()
-	for(var/datum/mind/rev_mind in head_revolutionaries)
-		var/turf/T = get_turf(rev_mind.current)
-		if((rev_mind) && (rev_mind.current) && (rev_mind.current.stat != DEAD) && rev_mind.current.client && T && is_station_level(T.z))
-			if(ishuman(rev_mind.current))
-				return FALSE
-	return TRUE
-
 //////////////////////////////////////////////////////////////////////
 //Announces the end of the game with all relavent information stated//
 //////////////////////////////////////////////////////////////////////
 /datum/game_mode/revolution/declare_completion()
-	if(finished == 1)
+	if(finished == REV_VICTORY)
 		SSticker.mode_result = "revolution win - heads killed"
 		to_chat(world, "<span class='redtext'>The heads of staff were killed or exiled! The revolutionaries win!</span>")
-	else if(finished == 2)
+	else if(finished == STATION_VICTORY)
 		SSticker.mode_result = "revolution loss - rev heads killed"
 		to_chat(world, "<span class='redtext'>The heads of staff managed to stop the revolution!</span>")
 	..()
 	return TRUE
 
 /datum/game_mode/proc/auto_declare_completion_revolution()
 	var/list/targets = list()
-	if(head_revolutionaries.len || GAMEMODE_IS_REVOLUTION)
+	if(length(rev_team?.members) || GAMEMODE_IS_REVOLUTION)
 		var/num_revs = 0
 		var/num_survivors = 0
-		for(var/mob/living/carbon/survivor in GLOB.alive_mob_list)
-			if(survivor.ckey)
-				num_survivors++
-				if(survivor.mind)
-					if((survivor.mind in head_revolutionaries) || (survivor.mind in revolutionaries))
-						num_revs++
+		for(var/mob/living/carbon/human/survivor in GLOB.player_list)
+			if(!istype(survivor) || survivor.stat == DEAD)
+				return
+			num_survivors++
+			if(survivor?.mind.has_antag_datum(/datum/antagonist/rev))
```

Comment:
```suggestion
			if(survivor.mind?.has_antag_datum(/datum/antagonist/rev))
```

Reply:
This will runtime if the survivor has no mind, I don't think we want the ?. being applied to checking of antag datum proc call

Reply:
You sure? We've already done a nullcheck on the survivor, this is performing the nullcheck on their mind, which I don't think we account for.

Reply:
Thats checking on the proc, ?. has to come before

Reply:
I don't know anymore, the docs confuse me. Now I think you're right on this one.